tools/RepoTasks/RepoTasks.RemoteWorker/AppDomainWorker.cs

```diff
@@ -0,0 +1,233 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace RepoTasks.RemoteWorker
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+    using System.Xml.Serialization;
+    using System.Management.Automation;
+    using Microsoft.WindowsAzure.Commands.Common.Attrubutes;
+
+    public class AppDomainWorker : MarshalByRefObject, IReflectionWorker
+    {
+        private Configuration _configuration;
+        private bool _onlyMarkedProperties;
+
+        public string BuildFormatPs1Xml(string assemblyPath, string[] cmdlets, bool onlyMarkedProperties)
+        {
+            _onlyMarkedProperties = onlyMarkedProperties;
+            string assemblyName;
+            var cmdletTypes = GetCmdletTypes(assemblyPath, cmdlets, out assemblyName);
+            if (cmdletTypes.Count == 0) return null;
+            _configuration = BuildXmlConfiguration(cmdletTypes);
+            var filename = assemblyName + ".generated.format.ps1xml";
+            return filename;
+        }
+
+        public void Serialize(string filepath)
+        {
+            Serialize(_configuration, filepath);
+        }
+
+        internal static IList<Type> GetCmdletTypes(string assemblyPath, string[] cmdlets, out string assemblyName)
+        {
+            if (string.IsNullOrEmpty(assemblyPath)) throw new ArgumentNullException(nameof(assemblyPath));
+            var assembly = Assembly.LoadFrom(assemblyPath);
+
+            assemblyName = assembly.GetName().Name;
+            var cmdletTypes = assembly
+                .ExportedTypes
+                .Where(
+                    type => {
+                        var attributes = type.GetCustomAttributes(
+                            typeof(CmdletAttribute),
+                            false);
+
+                        if (attributes.Length == 0) return false;
+                        var cmdletAttribute = (CmdletAttribute)attributes.First();
+                        if (cmdletAttribute == null) return false;
+                        if (!(cmdlets?.Length > 0)) return true;
+                        var attrCmdlet = $"{cmdletAttribute.VerbName}-{cmdletAttribute.NounName}";
+                        return cmdlets.Any(cmdlet => string.Equals(cmdlet, attrCmdlet, StringComparison.OrdinalIgnoreCase));
+                    })
+                .ToList();
+
+            return cmdletTypes;
+        }
+
+        private readonly ISet<string> _usedTypes = new HashSet<string>();
+
+        internal Configuration BuildXmlConfiguration(IEnumerable<Type> cmdletTypList)
+        {
+            var viewList = new List<View>();
+
+            foreach (var cmdletType in cmdletTypList)
+            {
+                var attributes = cmdletType.GetCustomAttributes(
```

Comment:
You might find these extensions useful:  https://github.com/Azure/azure-powershell/blob/preview/tools/StaticAnalysis/HelpAnalyzer/ReflectionExtensions.cs

Reply:
Not any required change, just FYI

Reply:
ok.